nuxeo-mongodb-directory/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -127,9 +135,21 @@ public DocumentModelList getEntries() throws DirectoryException {
     @Override
     public DocumentModel createEntry(Map<String, Object> fieldMap) throws DirectoryException {
         checkPermission(SecurityConstants.WRITE);
-        String id = String.valueOf(fieldMap.get(getIdField()));
-        if (hasEntry(id)) {
-            throw new DirectoryException(String.format("Entry with id %s already exists", id));
+        String id;
+        if (autoincrementId) {
+            Document filter = MongoDBSerializationHelper.fieldMapToBson(MONGODB_ID, directoryName);
+            Document update = new Document().append(MONGODB_INC,
```

Comment:
Agreed, but this can be done when this module will be moved to the platform, as we will share the same code between the modules _nuxeo-mongodb-directory_, _nuxeo-mongodb-seqgen_ and nuxeo-mongodb-repository_ and have this code in only one place.